src/registrar/management/commands/extend_expiration_dates.py

```diff
@@ -0,0 +1,209 @@
+        if limit_parse != 0:
+            domains_to_change_count = limit_parse
+            valid_domains = valid_domains[:limit_parse]
+
+        # Determines if we should continue code execution or not.
+        # If the user prompts 'N', a sys.exit() will be called.
+        self.prompt_user_to_proceed(extension_amount, domains_to_change_count)
+
+        for domain in valid_domains:
+            try:
+                is_idempotent = self.idempotence_check(domain, extension_amount)
+                if not disable_idempotence and not is_idempotent:
+                    self.update_skipped.append(domain.name)
+                    logger.info(f"{TerminalColors.YELLOW}" f"Skipping update for {domain}" f"{TerminalColors.ENDC}")
+                else:
+                    domain.renew_domain(extension_amount)
+                    self.update_success.append(domain.name)
+                    logger.info(
+                        f"{TerminalColors.OKCYAN}"
+                        f"Successfully updated expiration date for {domain}"
+                        f"{TerminalColors.ENDC}"
+                    )
+            # Catches registry errors. Failures indicate bad data, or a faulty connection.
+            except (RegistryError, KeyError, TransportError) as err:
+                self.update_failed.append(domain.name)
+                logger.error(
+                    f"{TerminalColors.FAIL}" f"Failed to update expiration date for {domain}" f"{TerminalColors.ENDC}"
+                )
+                logger.error(err)
+            except Exception as err:
+                self.log_script_run_summary(debug)
+                raise err
+        self.log_script_run_summary(debug)
+
+    # == Helper functions == #
+    def idempotence_check(self, domain: Domain, extension_amount):
+        """Determines if the proposed operation violates idempotency"""
+        # Because our migration data had a hard stop date, we can determine if our change
+        # is valid simply checking the date is within a valid range and it was updated
+        # in epp or not.
+        # CAVEAT: This is a workaround. A more robust solution would be a db flag
+        current_expiration_date = domain.registry_expiration_date
+        transition_domains = TransitionDomain.objects.filter(
+            domain_name=domain.name, epp_expiration_date=current_expiration_date
+        )
+
+        return transition_domains.count() > 0
+
+    def prompt_user_to_proceed(self, extension_amount, domains_to_change_count):
+        """Asks if the user wants to proceed with this action"""
+        TerminalHelper.prompt_for_execution(
+            system_exit_on_terminate=True,
+            info_to_inspect=f"""
+            ==Extension Amount==
+            Period: {extension_amount} year(s)
+
+            ==Proposed Changes==
+            Domains to change: {domains_to_change_count}
+            """,
+            prompt_title="Do you wish to proceed?",
```

Comment:
Content question: Feel free to ignore this if there are Figma wireframes for this phrase, but it may be more clarifying to the user to title the prompt "Do you wish to proceed with **these changes?**" or something in the last line that indicates what they will be doing by proceeding

Reply:
Great suggestion! Luckily for console related stuff we don't need to abide by a figma design, so I'm glad to change it